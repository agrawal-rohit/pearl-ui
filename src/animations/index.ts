import { useAnimationState } from "../hooks/useAnimationState";

/**
 * Custom hook for creating a fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from an opacity of 0 to an opacity of 1.
 * This creates a fade-in effect.
 *
 * @returns An animation state that creates a fade-in effect.
 */
export const useFade = () => {
  // Define the initial and final states for the fade animation.
  // The animation will start with an opacity of 0 (completely transparent) and end with an opacity of 1 (completely opaque).
  const fadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
    },
    to: {
      opacity: 1,
    },
  });

  // Return the fade animation state.
  return fadeAnimationState;
};

/**
 * Custom hook for creating a scale and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scale of 0.5 and opacity of 0 to a scale of 1 and opacity of 1.
 * This creates a combined effect of scaling and fading in.
 *
 * @returns An animation state that creates a scale and fade-in effect.
 */
export const useScaleFade = () => {
  // Define the initial and final states for the scale and fade animation.
  // The animation will start with a scale of 0.5 (half the original size) and opacity of 0 (completely transparent)
  // and end with a scale of 1 (original size) and opacity of 1 (completely opaque).
  const scaleFadeAnimationState = useAnimationState({
    from: {
      scale: 0.9,
      opacity: 0,
    },
    to: {
      scale: 1,
      opacity: 1,
    },
  });

  // Return the scale and fade animation state.
  return scaleFadeAnimationState;
};

/**
 * Custom hook for creating a slide animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 to a translateX of 0.
 * This creates a slide-in effect.
 *
 * @returns An animation state that creates a slide-in effect.
 */
export const useSlide = () => {
  // Define the initial and final states for the slide animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and end with a translateX of 0 (original position).
  const slideAnimationState = useAnimationState({
    from: {
      translateY: -100,
    },
    to: {
      translateY: 0,
    },
  });

  // Return the slide animation state.
  return slideAnimationState;
};

/**
 * Custom hook for creating a slide and fade animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a translateX of -100 and opacity of 0 to a translateX of 0 and opacity of 1.
 * This creates a combined effect of sliding and fading in.
 *
 * @returns An animation state that creates a slide and fade-in effect.
 */
export const useSlideFade = () => {
  // Define the initial and final states for the slide and fade animation.
  // The animation will start with a translateX of -100 (completely off-screen to the left) and opacity of 0 (completely transparent)
  // and end with a translateX of 0 (original position) and opacity of 1 (completely opaque).
  const slideFadeAnimationState = useAnimationState({
    from: {
      opacity: 0,
      translateY: 10,
    },
    to: {
      opacity: 1,
      translateY: 0,
    },
  });

  // Return the slide and fade animation state.
  return slideFadeAnimationState;
};

/**
 * Custom hook for creating a collapse animation state.
 * This hook uses the useAnimationState hook to create an animation state that transitions from a scaleY of 1 to a scaleY of 0.
 * This creates a collapse effect.
 *
 * @returns An animation state that creates a collapse effect.
 */
export const useCollapse = () => {
  // Define the initial and final states for the collapse animation.
  // The animation will start with a scaleY of 1 (original size) and end with a scaleY of 0 (collapsed).
  const collapseAnimationState = useAnimationState({
    from: {
      opacity: 0,
      height: 0,
    },
    to: {
      opacity: 1,
      height: 100,
    },
  });

  // Return the collapse animation state.
  return collapseAnimationState;
};
