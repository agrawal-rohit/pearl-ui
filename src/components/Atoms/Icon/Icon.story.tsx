import React from "react";
import { storiesOf } from "@storybook/react-native";
import Screen from "../Screen/Screen";
import Box from "../Box/Box";
import Icon from "./Icon";

storiesOf("Icon", module)
  .addDecorator((getStory) => <Screen>{getStory()}</Screen>)
  .add("Sizes", () => (
    <Box flexDirection="row" justifyContent="space-between">
      <Icon iconFamily="AntDesign" iconName="stepforward" size="s" />
      <Icon iconFamily="AntDesign" iconName="stepforward" size="m" />
      <Icon iconFamily="AntDesign" iconName="stepforward" size="l" />
      <Icon iconFamily="AntDesign" iconName="stepforward" size="xl" />
    </Box>
  ))
  .add("Overrides", () => (
    <Box flexDirection="row" justifyContent="space-between">
      <Icon
        iconFamily="AntDesign"
        iconName="stepforward"
        size="l"
        color="primary.500"
      />
    </Box>
  ));
