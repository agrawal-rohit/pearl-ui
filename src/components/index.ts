// Atoms
export { default as Box, BoxProps } from "./atoms/box/box";
export { default as Center } from "./atoms/center/center";
export { default as Collapse, CollapseProps } from "./atoms/collapse/collapse";
export { default as Divider, DividerProps } from "./atoms/divider/divider";
export { default as Fade, FadeProps } from "./atoms/fade/fade";
export { default as Grid, GridProps } from "./atoms/grid/grid";
export { default as Icon, IconProps } from "./atoms/icon/icon";
export {
  default as Pressable,
  PressableProps,
} from "./atoms/pressable/pressable";
export {
  default as ScaleFade,
  ScaleFadeProps,
} from "./atoms/scale-fade/scale-fade";
export {
  default as List,
  OrderedList,
  UnorderedList,
  ListProps,
} from "./atoms/list/list";
export { default as Screen, ScreenProps } from "./atoms/screen/screen";
export { default as Skeleton, SkeletonProps } from "./atoms/skeleton/skeleton";
export {
  default as SkeletonCircle,
  SkeletonCircleProps,
} from "./atoms/skeleton/skeleton-circle";
export {
  default as SkeletonText,
  SkeletonTextProps,
} from "./atoms/skeleton/skeleton-text";
export { default as Slide, SlideProps } from "./atoms/slide/slide";
export {
  default as SlideFade,
  SlideFadeProps,
} from "./atoms/slide-fade/slide-fade";
export { default as Spacer } from "./atoms/spacer/spacer";
export { default as Spinner, SpinnerProps } from "./atoms/spinner/spinner";
export {
  default as Stack,
  HStack,
  VStack,
  ZStack,
  HStackProps,
  VStackProps,
  StackProps,
} from "./atoms/stack/stack";
export { default as Text, TextProps } from "./atoms/text/text";

// Molecules
export { default as Avatar, AvatarProps } from "./molecules/avatar/avatar";
export {
  default as AvatarGroup,
  AvatarGroupProps,
} from "./molecules/avatar/avatar-group";
export { default as Badge, BadgeProps } from "./molecules/badge/badge";
export {
  default as withBadge,
  WithBadgeOptions,
} from "./molecules/badge/withBadge";
export { default as Button, ButtonProps } from "./molecules/button/button";
export {
  default as ButtonGroup,
  ButtonGroupProps,
} from "./molecules/button/button-group";
export {
  default as CheckBox,
  CheckBoxProps,
} from "./molecules/checkbox/checkbox";
export {
  default as CheckBoxGroup,
  CheckBoxGroupProps,
} from "./molecules/checkbox/checkbox-group";
export {
  default as IconButton,
  IconButtonProps,
} from "./molecules/icon-button/icon-button";
export { default as Image, ImageProps } from "./molecules/image/image";
export { default as Input, InputProps } from "./molecules/input/input";
export {
  default as PinInput,
  PinInputProps,
} from "./molecules/pin-input/pin-input";
export {
  default as Progress,
  ProgressProps,
} from "./molecules/progress/progress";
export { default as Radio, RadioProps } from "./molecules/radio/radio";
export {
  default as RadioGroup,
  RadioGroupProps,
} from "./molecules/radio/radio-group";
export { default as Switch, SwitchProps } from "./molecules/switch/switch";
export {
  default as Textarea,
  TextareaProps,
} from "./molecules/textarea/textarea";
export { default as Video, VideoProps } from "./molecules/video/video";
export {
  default as Accordion,
  AccordionProps,
  AccordionItemProps,
  AccordionButtonProps,
  AccordionPanelProps,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
} from "./molecules/accordion/accordion";
export { default as Alert, AlertProps } from "./molecules/alert/alert";
