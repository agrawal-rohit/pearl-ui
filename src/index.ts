export { default as Box } from "./components/Box/Box";
export { default as Button } from "./components/Button/Button";
export { default as Text } from "./components/Text/Text";
