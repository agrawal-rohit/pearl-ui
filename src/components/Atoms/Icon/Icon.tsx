import React from "react";
import {
  backgroundColor,
  BackgroundColorProps,
  color,
  ColorProps,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from "../../../theme/src/styleFunctions";
import { RNStyle, StyleFunctionContainer } from "../../../theme/src/types";
import * as ExpoIcons from "@expo/vector-icons";
import { useComponentConfig } from "../../../hooks/useComponentConfig";

const iconStyleFunctions = [
  color,
  backgroundColor,
  spacing,
  layout,
] as StyleFunctionContainer[];

type IconProps = ColorProps &
  BackgroundColorProps &
  SpacingProps &
  LayoutProps & {
    /** Icon family that contains the icon you want to use  */
    iconFamily:
      | "AntDesign"
      | "Entypo"
      | "EvilIcons"
      | "Feather"
      | "FontAwesome"
      | "FontAwesome5"
      | "Fontisto"
      | "Foundation"
      | "Ionicons"
      | "MaterialCommunityIcons"
      | "MaterialIcons"
      | "Octicons"
      | "SimpleLineIcons"
      | "Zocial";
    /** Name of the icon as given in it's respective icon family */
    iconName: string;
    /** The size of the icon */
    size?: string;
    /** The accessibility label of the icon */
    accessibilityLabel?: string;
    style?: RNStyle;
  };

/** he `Icon` component can used to add Expo Icons to your app and customize them using style props. */
const Icon: React.FC<IconProps> = ({
  iconFamily,
  iconName,
  size = "m",
  accessibilityLabel = null,
  ...props
}) => {
  const componentSpecificProps = useComponentConfig(
    "Icon",
    props,
    {
      size: size,
    },
    iconStyleFunctions
  );

  const IconToUse = ExpoIcons[iconFamily];

  return (
    <IconToUse
      accessible={true}
      accessibilityLabel={
        accessibilityLabel ? accessibilityLabel : `${iconName} Icon`
      }
      name={iconName}
      {...componentSpecificProps}
    ></IconToUse>
  );
};

export default Icon;
