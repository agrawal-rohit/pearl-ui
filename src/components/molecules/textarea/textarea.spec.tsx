import React from "react";
import Textarea from "./textarea";
import { fireEvent, render } from "@testing-library/react-native";
import { ThemeProvider } from "../../../theme/src/theme-context";
import Icon from "../../atoms/icon/icon";

jest.useFakeTimers();

describe("Molecules/Textarea", () => {
  it("passes the snapshot test for basic setup", async () => {
    const tree = await render(
      <ThemeProvider>
        <Textarea placeholder="This is a placeholder" />
      </ThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different sizes", async () => {
    const tree = await render(
      <ThemeProvider>
        <Textarea size="xs" />
        <Textarea size="s" />
        <Textarea size="m" />
        <Textarea size="l" />
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("passes the snapshot test for different variants", async () => {
    const tree = await render(
      <ThemeProvider>
        <Textarea variant="filled" />
        <Textarea variant="outline" />
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("updates styles correctly when in an error state ", async () => {
  //   const errorBackgroundColor = "neutral.50";
  //   const errorBorderColor = "danger.800";

  //   const { getByTestId } = render(
  //     <ThemeProvider>
  //       <Textarea
  //         testID="inputField"
  //         isInvalid
  //         errorBackgroundColor={errorBackgroundColor}
  //         errorBorderColor={errorBorderColor}
  //       />
  //     </ThemeProvider>
  //   );

  //   const inputContainer = getByTestId("inputFieldContainer");
  //   expect(inputContainer.props.style.borderColor).toEqual(
  //     (baseTheme.palette.danger as any)["800"]
  //   );
  //   expect(inputContainer.props.style.backgroundColor).toEqual(
  //     (baseTheme.palette.neutral as any)["50"]
  //   );
  // });

  it("passes the snapshot test when using style props", async () => {
    const tree = await render(
      <ThemeProvider>
        <Textarea
          backgroundColor="primary.500"
          borderColor="neutral.400"
          borderWidth={2}
          isFullWidth
          placeholderTextColor={undefined}
        />
      </ThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("captures onChangeText event correctly", () => {
    const value = "start value";
    const onChangeText = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider>
        <Textarea
          placeholder="Enter value"
          onChangeText={onChangeText}
          value={value}
          testID="inputField"
        />
      </ThemeProvider>
    );

    const input = getByTestId("inputField");
    fireEvent.changeText(input, "new value");

    expect(onChangeText).toBeCalledWith("new value");
  });

  it("doesn't update value when the input is disabled ", async () => {
    const onChangeText = jest.fn();

    const { getByTestId, queryByText } = await render(
      <ThemeProvider>
        <Textarea onChangeText={onChangeText} isDisabled testID="inputField" />
      </ThemeProvider>
    );

    const input = getByTestId("inputField");
    fireEvent.changeText(input, "new value");

    expect(queryByText("new value")).toBeNull();
  });
});
