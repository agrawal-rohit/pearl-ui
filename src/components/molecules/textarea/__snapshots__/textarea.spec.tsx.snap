// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Textarea passes the snapshot test for basic setup 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityLabel="This is a placeholder"
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="This is a placeholder"
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
</View>
`;

exports[`Molecules/Textarea passes the snapshot test for different sizes 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 4,
        "paddingVertical": 4,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 4,
          "paddingVertical": 4,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 4,
        "paddingVertical": 4,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 13.799999999999999,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 20,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 4,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 4,
        "paddingVertical": 6,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 4,
          "paddingVertical": 6,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 4,
        "paddingVertical": 6,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 13.799999999999999,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 20,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 6,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 18.4,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 26,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
</View>
`;

exports[`Molecules/Textarea passes the snapshot test for different variants 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
      variant="filled"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#C5CEE0",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#C5CEE0",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#C5CEE0",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
      variant="outline"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
</View>
`;

exports[`Molecules/Textarea passes the snapshot test when using style props 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    animate={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    animatedStyle={
      {
        "value": {
          "alignSelf": "flex-start",
          "backgroundColor": "#EDF1F7",
          "borderColor": "#E4E9F2",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 100,
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
          "width": "100%",
        },
      }
    }
    collapsable={false}
    style={
      {
        "alignSelf": "flex-start",
        "backgroundColor": "#EDF1F7",
        "borderColor": "#E4E9F2",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 8,
        "width": "100%",
      }
    }
    transition={
      {
        "duration": 100,
      }
    }
  >
    <TextInput
      accessibilityState={
        {
          "disabled": false,
          "selected": false,
        }
      }
      allowFontScaling={true}
      editable={true}
      inputMode="text"
      multiline={true}
      numberOfLines={4}
      onBlur={[Function]}
      onChange={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#8F9BB3"
      selectionColor="#6A7BFF"
      style={
        [
          {
            "marginHorizontal": 4,
          },
          {
            "color": "#1A2138",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16.099999999999998,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {
            "flex": 1,
          },
        ]
      }
      textAlignVertical="top"
    />
    <View
      animate={
        {
          "alignItems": "center",
          "alignSelf": "center",
        }
      }
      animatedStyle={
        {
          "value": {
            "alignItems": "center",
            "alignSelf": "center",
          },
        }
      }
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        }
      }
      transition={
        {
          "dampingRatio": 1,
          "duration": 100,
          "type": "spring",
        }
      }
    />
  </View>
</View>
`;
